core/src/main/java/com/datastrato/gravitino/storage/kv/KvEntityStore.java

```diff
@@ -390,6 +407,7 @@ public boolean delete(NameIdentifier ident, EntityType entityType, boolean casca
                     .build());
           }
 
+          deleteUserEntitiesIfNecessary(ident, entityType);
           return transactionalKvBackend.delete(dataKey);
         });
   }
```

Comment:
@yuqi1129 please help to check this part. I don't think the change is enough to support `User`.

Reply:
@qqqttt123 
1. If a catalog and a user share the same entity, Gravitino will receive an error, so we need to take actions to prevent this from happening.
2. You may also need to implement rational entity stores. 
3.

Reply:
1. How to prevent it?
2. I prefer doing it in another pr.

Reply:
> How to prevent it?

We need to consider it any way, let me think about it also.

Reply:
@qqqttt123 
We'd better add some comments about it, If the name of a user and catalog is the same, they will share the same id in the id-name mapping, If we drop any one( the user or the catalog), the ID of it may also be removed( though we will not abandon the id of the name when dropping an entity), so I suggest we need to add some comment about it.

Reply:
Comment doesn't solve anything, this should have a mechanism to fix.

Reply:
There is no issues in current implement, yuqi just worried about wrong modification will happen in the future. So he suggests to add some comments.

Reply:
@jerryshao 
Currently, we won't remove the id-name mapping data when dropping an entity to reuse it. I'm not sure if we will modify the resuing policy in the future when the data is too large. If it won't be changed in the future, it is okay with the current changes (for me, I would be hesitant to change it if there is no special reason). 

Another potential issue is that renaming may not go as planned. For instance, the catalog 'A' and user `A` share the same id 1, when we rename catalog `A` to `B`, then the mapping then become B to 1 and user `A` can't be found. 

What do you think about it? After discussing with @qqqttt123, I think we need to confirm the following things:
1. Is it feasible to store user, role, and group information in a metalake?
2. Is it acceptable to share IDs between catalogs and users?
3. If the answers to the first two questions are affirmative, we may need to modify the current storage format and change the id-name mapping mechanism to support it.

Reply:
In my thought,
1. Metalake is a tenant concept us. We have users, groups and roles under the metalake. They will be isolated among metalakes.

Reply:
What I want is that the semantic is correct to support different operations, no matter what underlying mapping you use.

Besides, I don't think the storage hierarchy should be directly mapping the semantic layer. Like user, it is a concept under the metalake. But we can store them in a system catalog/table, we don't have to store it directly under metalake.

Reply:
> Another potential issue is that renaming may not go as planned. For instance, the catalog 'A' and user A share the same id 1, when we rename catalog A to B, then the mapping then become B to 1 and user A can't be found.

This problem should also exist for other entities, it should be fixed. For example, renaming table from A to B, then the fileset A cannot be found either.

Reply:
You should never assume that there's only one entity type under the specific namespace.

Reply:
Do we need to wait for @yuqi1129 's fix or we can merge it beforehand?